interface/ceed-preconditioning.c

```diff
@@ -1160,132 +1262,172 @@ int CeedOperatorAssemblyDataCreate(Ceed ceed, CeedOperator op, CeedOperatorAssem
       }
     }
   }
-  (*data)->num_eval_mode_out = num_eval_mode_out;
-  (*data)->eval_mode_out     = eval_mode_out;
-  CeedCall(CeedBasisReferenceCopy(basis_out, &(*data)->basis_out));
+  (*data)->num_output_components = offset;
+  (*data)->num_eval_modes_out    = num_eval_modes_out;
+  (*data)->eval_modes_out        = eval_modes_out;
+  (*data)->eval_mode_offsets_out = eval_mode_offsets_out;
+  (*data)->num_active_bases      = num_active_bases;
 
   return CEED_ERROR_SUCCESS;
 }
 
 /**
   @brief Get CeedOperator CeedEvalModes for assembly
 
-  @param[in]  data              CeedOperatorAssemblyData
-  @param[out] num_eval_mode_in  Pointer to hold number of input CeedEvalModes, or NULL
-  @param[out] eval_mode_in      Pointer to hold input CeedEvalModes, or NULL
-  @param[out] num_eval_mode_out Pointer to hold number of output CeedEvalModes, or NULL
-  @param[out] eval_mode_out     Pointer to hold output CeedEvalModes, or NULL
+  @param[in]  data                  CeedOperatorAssemblyData
+  @param[out] num_active_bases      Number of active bases
+  @param[out] num_eval_mode_in      Pointer to hold array of numbers of input CeedEvalModes, or NULL
+  @param[out] eval_mode_in          Pointer to hold arrays of input CeedEvalModes, or NULL
+  @param[out] eval_mode_offsets_in  Pointer to hold arrays of input offsets at each quadrature point
+  @param[out] num_eval_mode_out     Pointer to hold array of numbers of output CeedEvalModes, or NULL
+  @param[out] eval_mode_out         Pointer to hold arrays of output CeedEvalModes, or NULL
+  @param[out] eval_mode_offsets_out Pointer to hold arrays of output offsets at each quadrature point
+  @param[out] num_output_components Total number of output components at each quadrature point
 
   @return An error code: 0 - success, otherwise - failure
```

Comment:
I think this man page would benefit from an explanation of the data model. Does `num_active_bases` apply to in or out? How is `num_output_components` counted across the various fields?

Reply:
Agreed - the names are less obvious in the light of Monday morning.